src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -1472,9 +1472,15 @@ def _update_vm_nics(cmd, vm, nics, primary_nic):
 
 
 # region VirtualMachines RunCommand
-def run_command_invoke(cmd, resource_group_name, vm_name, command_id, scripts=None, parameters=None):
+def run_command_invoke(cmd, resource_group_name, vm_vmss_name, command_id, scripts=None, parameters=None, instance_id=None):  # pylint: disable=line-too-long
     RunCommandInput, RunCommandInputParameter = cmd.get_models('RunCommandInput', 'RunCommandInputParameter')
 
+    # ensure helper is used properly.
+    if cmd.name.startswith("vmss ") and not instance_id:
+        raise CLIError("Internal Error: helper method called for vmss command without instance_id")
+    elif cmd.name.startswith("vm ") and instance_id:
+        raise CLIError("Internal Error: helper method called for vm command with instance_id")
```

Comment:
Are these really going to happen? Or more like a dev ASSERT?

Reply:
Since the test coverage is pretty good, I suggest just remove them, unless they represent errors which could be triggered by user errors

Reply:
I think this is more like a dev assert... so should I convert it to an assert or remove them?